backend/app/anti_recommenders/open_ai/open_ai_normal_anti_recommender.py

```diff
@@ -1,50 +1,48 @@
-from langchain_openai import OpenAI
+from collections.abc import Generator
+from typing import Any
+
 from langchain.prompts import PromptTemplate
-from langchain.schema.runnable import RunnablePassthrough, RunnableSerializable
 from langchain.schema import StrOutputParser
+from langchain.schema.runnable import RunnablePassthrough, RunnableSerializable
+from langchain_openai import OpenAI
 
-from collections.abc import Generator
 from app.anti_recommenders.open_ai.open_ai_anti_recommender import OpenAiAntiRecommender
-from app.models.anti_recommendation.anti_recommendation import AntiRecommendation
+from app.models.anti_recommendation import AntiRecommendation
 
 
 class OpenAiNormalAntiRecommender(OpenAiAntiRecommender):
     """
     A subclass of OpenAiAntiRecommender that relies solely on
-    the large language model's parametric knowledge to generate anti-recommendations.
+    the large language model's parametric knowledge to generate AntiRecommendations.
     """
 
     def __init__(self) -> None:
         super().__init__()
-        self.open_ai_normal_chain: RunnableSerializable | None = None
 
-    def _build_chain(self) -> None:
+    def _build_chain(self) -> RunnableSerializable[Any, str]:
```

Comment:
Ruff should flag use of `Any` in favor of a more specific return type.

Reply:
I couldn't find the precise type of this chain so I changed it to **RunnableSerializable**. I will continue looking through LangChain's docs to find a more specific type.